Source/WebCore/page/Navigation.cpp

```diff
@@ -972,69 +980,69 @@ Navigation::DispatchResult Navigation::innerDispatchNavigateEvent(NavigationNavi
             else if (callbackResult.type() == CallbackResultType::ExceptionThrown) {
                 // FIXME: We need to keep around the failure reason but the generated handleEvent() catches and consumes it.
                 auto promiseAndWrapper = createPromiseAndWrapper(*document);
-                promiseAndWrapper.second->reject(ExceptionCode::TypeError);
+                Ref { promiseAndWrapper.second }->reject(ExceptionCode::TypeError);
                 promiseList.append(WTFMove(promiseAndWrapper.first));
             }
         }
 
         if (promiseList.isEmpty()) {
             auto promiseAndWrapper = createPromiseAndWrapper(*document);
-            promiseAndWrapper.second->resolveWithCallback([](JSDOMGlobalObject&) {
+            Ref { promiseAndWrapper.second }->resolveWithCallback([](JSDOMGlobalObject&) {
                 return JSC::jsUndefined();
             });
             promiseList.append(WTFMove(promiseAndWrapper.first));
         }
 
         // FIXME: this emulates the behavior of a Promise wrapped around waitForAll, but we may want the real
         // thing if the ordering-and-transition tests show timing related issues related to this.
-        protectedScriptExecutionContext()->checkedEventLoop()->queueTask(TaskSource::DOMManipulation, [this, promiseList, abortController, document, apiMethodTracker]() {
-            waitForAllPromises(promiseList, [abortController, document, apiMethodTracker, weakThis = WeakPtr { *this }]() mutable {
+        protectedScriptExecutionContext()->checkedEventLoop()->queueTask(TaskSource::DOMManipulation, [weakThis = WeakPtr { this }, promiseList, abortController, document, apiMethodTracker]() {
+            waitForAllPromises(promiseList, [abortController, document, apiMethodTracker, weakThis]() mutable {
                 if (!weakThis || abortController->signal().aborted() || !document->isFullyActive() || !weakThis->m_ongoingNavigateEvent)
                     return;
 
-                RefPtr strongThis = weakThis.get();
+                RefPtr protectedThis = weakThis.get();
```

Comment:
Normally we want to null check `protectedThis`, not `weakThis`. It's not a big deal in single threaded code but still not good practice.

Reply:
Fixed